python/cuml/neighbors/nearest_neighbors.pyx

```diff
@@ -329,22 +324,10 @@ class NearestNeighbors(Base):
             False
         )
 
-        I_ndarr = I_ndarr.reshape((N, n_neighbors))
-        D_ndarr = D_ndarr.reshape((N, n_neighbors))
-
-        if isinstance(X, cudf.DataFrame):
-            inds = cudf.DataFrame.from_gpu_matrix(I_ndarr)
-            dists = cudf.DataFrame.from_gpu_matrix(D_ndarr)
-
-        elif isinstance(X, np.ndarray):
-            inds = np.asarray(I_ndarr)
-            dists = np.asarray(D_ndarr)
-
-        del I_ndarr
-        del D_ndarr
         del X_m
```

Comment:
I believe these explicit deletes can be removed. They should be deleted anyways when their references go out of scope. Tagging @dantegd because I can't recall if we were originally doing this as a result of some numba bug.